media-libs/nvidia-vaapi-driver/nvidia-vaapi-driver-0.0.12.ebuild

```diff
@@ -20,3 +20,13 @@ RDEPEND="media-libs/gst-plugins-bad
 DEPEND="${RDEPEND}
 	>=media-libs/nv-codec-headers-11.1.5.1"
 BDEPEND="virtual/pkgconfig"
+
+pkg_postinst() {
+	# Source: https://github.com/elFarto/nvidia-vaapi-driver/blob/master/src/backend-common.c#L13
+	ewarn "If vaapi drivers fail to load, then make sure that you are"
+	ewarn "passing the correct parameters to the kernel."
+	ewarn "nvidia_drm.modeset should be set to 1."
+
+	ewarn "Check the wiki page for more information: "
+	ewarn "https://wiki.gentoo.org/wiki/VAAPI"
```

Comment:
`elog` should be enough here. I also wonder should these messages be printed on first install only, or everytime the library is updated? I don't mind either way.

Reply:
Maybe look into whether drm modeset is enabled already?

Reply:
> Maybe look into whether drm modeset is enabled already?

How should/could it be achieved? Is there an ebuild specific command or are you suggesting to use a shell command? 

Thanks in advance!

Reply:
> `elog` should be enough here. I also wonder should these messages be printed on first install only, or everytime the library is updated? I don't mind either way.

I felt like printing the elog message every time the package is installed would help systems where an update happens to break something. A "just in case" kind of message. A bit like the [spotify ebuild message](https://github.com/gentoo/gentoo/blob/master/media-sound/spotify/spotify-1.2.40.ebuild#L127-L129.)

However, using `DOC_CONTENTS` would also make sense! 

Would love to get some feedback.

Reply:
> How should/could it be achieved? Is there an ebuild specific command or are you suggesting to use a shell command?

Had a quick look hoping it'd show up somewhere in /proc nvidia files, but haven't found it (not that I looked much). *Could* do poor heuristics to figure it out (like check /proc/cmdline + /etc/modprobe.d) if really wanted to but I don't think that's great and may not always be right.

Reply:
`cat /sys/module/nvidia_drm/parameters/modeset` but it actually requires root access, not sure if that works from all ebuild phases.

Reply:
Oh right, forgot despite I used to check ertm on xpadneo that way (not that need to anymore as you know, so that's gone)

wrt root, warning messages are typically done in pkg_postinst and that typically is root. So file will only be unreadable: 1. nvidia_drm is not loaded (I actually use X+nvidia without nvidia_drm loaded myself), nvidia is not loaded yet at all (e.g. chroot, different system), prefix system without root

Think it wouldn't be a big deal to be noisy when couldn't read the file though, so could silence errors and just do (untested):
```
	if [[ $(</sys/module/nvidia_drm/parameters/modeset) != Y ]] 2>/dev/null; then
		<warn>
	fi
```
Should only skip if it was readable and found `Y` in it.